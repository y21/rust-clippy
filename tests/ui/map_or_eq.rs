#![allow(unused)]
#![warn(clippy::map_or_eq)]

fn potential_side_effect<T>() -> T {
    panic!();
}

fn main() {
    let opt = None::<i32>;
    let _ = opt.map_or(false, |val| val == 5); // lint
    let _ = potential_side_effect::<Option<i32>>().map_or(false, |val| val == 5); // lint
    let _ = opt.map_or(false, |val| val != potential_side_effect()); // don't lint
}
