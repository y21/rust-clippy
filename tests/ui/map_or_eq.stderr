error: this can be written more concisely using `==`
  --> $DIR/map_or_eq.rs:10:13
   |
LL |     let _ = opt.map_or(false, |val| val == 5); // lint
   |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: try: `opt == Some(5)`
   |
   = note: `-D clippy::map-or-eq` implied by `-D warnings`

error: this can be written more concisely using `==`
  --> $DIR/map_or_eq.rs:11:13
   |
LL |     let _ = potential_side_effect::<Option<i32>>().map_or(false, |val| val == 5); // lint
   |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: try: `potential_side_effect::<Option<i32>>() == Some(5)`

error: aborting due to 2 previous errors

